### DISCUSSION TITLE
Why don't provide a container image instead?

### DISCUSSION THREAD

**Discussion by jockexa** [swedenconnect/bankid-saml-idp#204] [General]
I've been struggling with this project for a bit now to be honest..

For me, that does not have a background from development, it's kind of strange to first have to build everything myself from src.
Just to get the .jar need for starting the service and just by then realize that even the "local"-mode is not configured to run as a container since the redis-server is hardcoded against local.dev.swedenconnect.se and not the name of the service from compose...
If that's fixed then the TLS-certificates does not match so that have to be adjusted as well etc. etc. 

So unfortunately (in my opinion) there is no easy-way to actually deploy and test this project yet.
In my opinion it would be so much easier if this project included some Dockerfile and docker-compose for the other services need to spin up a local testing and evaluation environment. 
But maybe there is a reason why this project hasn't been containerized yet?

**Comment by felix-hellman** [comment]
The local profile is for local development. I.e. Running in your IDE. If you want to deploy the service I would recommend you to write a configuration for that environment.

To build a Docker image of the application I would recommend starting out with reading the documententation.

https://github.com/swedenconnect/bankid-saml-idp/blob/main/docs/development.md#using-docker

If you want to change any configuration I would recommend you to read the documententation about configuration and ask more pointed questions about what configuration you believe not to be working.

Closing this as no question was asked that could not be answered by reading the documententation.
